src/components/shared/uicomponents/CustomerCommissionList.js

```diff
@@ -0,0 +1,36 @@
+import {DOMAIN_URL} from '../../../utils/Constants';
+
+import React from 'react';
+import axios from 'axios';
+
+export default class CustomerCommissionList extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            loading: true,
+            customerCommissions: [],
+        };
+    }
+
+    componentDidMount() {
+        axios.get(`${DOMAIN_URL}/api/v1/user/${this.props.userId}/customer-commissions`)
+        .then((response) => {
+            if (response.status === 200 || response.status === 304) {
+                this.setState({ customerCommissions: response.data.commissions });
+            }
+            this.setState({loading: false});
+        }).catch((error) => {
+            console.log(error);
```

Comment:
When theres an error, how should we convey it to the user? Note L21 will never be run so if the component errors out during the fetch it'll forever be stuck in a loading state

Reply:
should i have an error message then? and i should also set the loading state to false in the .catch block right?

Reply:
we can look at displaying an error message in the future, but now at least what we should do is set the loading state to false and some error state to true